src/vs/workbench/contrib/files/browser/fileCommands.ts

```diff
@@ -408,6 +413,39 @@ function revealResourcesInOS(resources: URI[], windowsService: IWindowsService,
 	}
 }
 
+KeybindingsRegistry.registerCommandAndKeybindingRule({
+	id: OPEN_WITH_COMMAND_ID,
+	weight: KeybindingWeight.WorkbenchContrib,
+	when: EditorContextKeys.focus.toNegated(),
+	handler: async (accessor: ServicesAccessor, resource: URI | object) => {
+		const editorService = accessor.get(IEditorService);
+		const resources = getMultiSelectedResources(resource, accessor.get(IListService), editorService);
+		const targetResource = resources[0];
+		if (!targetResource) {
+			return;
+		}
+
+		const resourceInput: IResourceInput = { resource: targetResource };
+		const bigInput = editorService.createInput(resourceInput);
+		if (!(bigInput instanceof FileEditorInput)) {
+			return;
+		}
+
+		const preferredEditors = Registry.as<IEditorRegistry>(EditorExtensions.Editors).getPreferredEditorsForResource(targetResource);
+		const pick = await accessor.get(IQuickInputService).pick(
+			preferredEditors.map((editorDescriptor): IQuickPickItem => ({
+				label: editorDescriptor.getName(),
+				id: editorDescriptor.getId()
+			})),
+			{});
+
+		if (pick) {
+			bigInput.setPreferredCustomEditor(pick.id!);
```

Comment:
I believe inputs are lightweight and short lived and therefore this has no effect?